src/Manifolds.jl

```diff
@@ -104,6 +104,22 @@ using FiniteDifferences
 using UnsafeArrays
 using Einsum: @einsum
 
+"""
+    Field
+
+An abstract type to represent the field matrix manifolds are build upon,
+following the idea of [TensorKit](https://github.com/Jutho/TensorKit.jl)
+in order to hace concrete field types to dispatch on. The two most common
+field types are `RealNumbers` (`ℝ` for short) and `ComplexNumbers` (`ℂ`).
+"""
+abstract type Field end
```

Comment:
Maybe we should call this `AbstractField`?

Reply:
I don't mind and it would be consistent with our naming scheme, you're right.